.github/workflows/rebase.yaml

```diff
@@ -0,0 +1,12 @@
+name: rebase pull request
+on:
+  push:
+    branches:
+    - "main"
+jobs:
+  rebase:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: peter-evans/rebase@v1
+        with:
+          base: main
```

Comment:
For more context, it looks like you can achieve some of this using native branch protection policies on the default branch (main in this case) and we should enable that, but it's still a manual effort in terms of rebasing content. Other tooling around rebasing PRs when changes land in master have different stories specifically around forks opened PRs varying from it's not supported, to you need to have a custom PAT token configured for the repository, vs. this one which just requires allowing maintainers to edit the pull request content.

Reply:
Link to the allowing 'edits from maintainers' documentation: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork.

Reply:
Is this the setting you're talking about in the first comment?

<img width="836" alt="Screen Shot 2021-11-11 at 3 49 48 PM" src="https://user-images.githubusercontent.com/54378333/141367096-b9856f31-4aa8-4ea6-9ce7-69c7dfe2fc1d.png">

Reply:
Yeah, those are the checks. You'll need to manually define the required checks when I poking around those options earlier today, but I think that should give us the requisite coverage for 1.) automatic rebasing of PRs once changes land in the main branch and 2.) decrease the chance of regressing main for open PRs.

Reply:
Got it. I believe that this should work once we add this rebase check you're merging so this LGTM.